app/models/manageiq/providers/redhat/infra_manager/vm.rb

```diff
@@ -102,4 +107,103 @@ def has_required_host?
   def self.calculate_power_state(raw_power_state)
     POWER_STATES[raw_power_state] || super
   end
+
+  #
+  # This method is executed in the UI worker.
+  # It adds a task to calculate the URL of oVirt UI for this virtual machine.
+  #
+  # @return [Integer] The identifier of the task.
+  #
+  def queue_generate_ui_url
+    # Create the task options:
+    task_options = {
+      action: "Generate oVirt UI URL for VM '#{name}'"
+    }
+
+    # Create the queue options:
+    queue_options = {
+      class_name: self.class.name,
+      instance_id: id,
```

Comment:
When I did the initial prototype of this I placed the code in the `Vm` class because the URL was (in theory) for a specific virtual machine. But if we are going to go to the virtual machines tab, not to a specific virtual machine, it may make more sense to place this in the `InfraManager` class instead (maybe as a mixin).